components/dashboard/src/user-settings/Account.tsx

```diff
@@ -49,14 +53,18 @@ export default function Account() {
                 return;
             }
         } else {
-            profileState.email = User.getPrimaryEmail(user) || "";
+            profileState.email = getPrimaryEmail(user) || "";
         }
 
-        const updatedUser = User.setProfile(user, profileState);
-        setUser(updatedUser);
-        getGitpodService().server.updateLoggedInUser(updatedUser);
+        await updateUser.mutateAsync({
```

Comment:
it seems before we won't wait, not sure what is desired here

Reply:
reload should be called after update, so it makes sence. But not sure if we need to do it, because udpate will respond an User, we could reuse it? Or not in this PR?